runtime/lua/provider/python/health.lua

```diff
@@ -343,7 +343,7 @@ function M.check()
     end
   end
 
-  if python_exe == '' and pyname ~= '' then
+  if pyname and python_exe == '' and pyname ~= '' then
```

Comment:
I didn't study the code flow very deeply but this works correctly for the "neovim module installed" and "not installed" cases. @dundargoc

Reply:
Works for me